File: app/src/main/java/com/bakkenbaeck/sol/service/AlarmReceiver.java
package com.bakkenbaeck.sol.service;


import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

public class AlarmReceiver extends BroadcastReceiver {

    @Override
    public void onReceive(final Context context, final Intent intent) {
        final Intent service = new Intent(context, SunsetService.class);
        service.putExtra(SunsetService.EXTRA_SHOW_NOTIFICATION, true);
        context.startService(service);
    }
}

File: app/src/main/java/com/bakkenbaeck/sol/service/AlarmReceiver.kt
package com.bakkenbaeck.sol.service


import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import com.bakkenbaeck.sol.BaseApplication

class AlarmReceiver : BroadcastReceiver() {

    override fun onReceive(context: Context, intent: Intent) {
        BaseApplication.instance.refreshLocation(true)
    }
}
